rescan.go

```diff
@@ -482,24 +444,28 @@ func rescan(chain ChainSource, options ...RescanOption) error {
 	//
 	// TODO(wilmer): refactor this and handleBlockDisconnected into their
 	// own methods.
-	handleBlockConnected := func(ntfn *blockntfns.Connected) error {
+	handleBlockConnected := func(ntfn *blockntfns.Connected, isRetry bool) error {
```

Comment:
Perhaps it's time to extract this closure (and the disconnect one) to a distinct method?

Reply:
Both closures currently depend on a lot of other internal state within the `rescan` method, so I'd rather defer doing so until we refactor the `Rescan` struct itself when moving towards a `rescan` package.

Reply:
There is already the struct `Rescan` that could hold that internal state. `Rescan` is a single use object anyway. :+1: on extracting the closures. Preferably in a commit before the actual fix is implemented.